debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/RecordsSnapshotProducer.java

```diff
@@ -357,13 +355,7 @@ private Object valueForColumn(ResultSet rs, int colIdx, ResultSetMetaData metaDa
             logger.trace("Type is: {}", type);
 
             if (type.isArrayType()) {
-                Array array = rs.getArray(colIdx);
-
-                if (array == null) {
-                    return null;
-                }
-
-                return Arrays.asList((Object[]) array.getArray());
+                return rs.getArray(colIdx);
```

Comment:
I don't quite understand the fix? Why does it help to return the `PgArray`?

Reply:
@gunnarmorling The problem was that in case of streaming the array of unknow datatypes was handled as unknown datattypes so in both cases wal2json and protobuf was sending the content as an octet stream - typically the string rpresantion of the array.
In case of snapshot we hadnle array as array and convert it to list. In value converter we are then not able to reconstruct the string/binary represantion of the array from the array list. Thus we pass PgArray and use its array content or its binary/string serialized form.